src/kani-compiler/src/codegen_cprover_gotoc/codegen/place.rs

```diff
@@ -550,3 +584,39 @@ impl<'tcx> GotocCtx<'tcx> {
         arr.member("0", &self.symbol_table).index_array(idx)
     }
 }
+
+/// A convenience macro that unwraps a `Result<ProjectPlace<'tcx>,
+/// Err<UnimplementedData>` if it is Ok and returns an `codegen_unimplemented`
+/// expression otherwise
+#[macro_export]
+macro_rules! unwrap_or_codegen_unimplemented {
```

Comment:
This macro affects control flow where it is called; we should make sure to call that our clearly in the name.

Reply:
I added return in the macro name: `unwrap_or_return_codegen_unimplemented`. Let me know if you have other suggestions.